api/models/status.go

```diff
@@ -0,0 +1,67 @@
+package models
+
+import (
+	"fmt"
+	"time"
+)
+
+type Defaulter interface {
+	Default()
+}
+
+func NewResourceCleanupStatus(state ResourceCleanupStatusState, m string, s ...string) *ResourceCleanupStatus {
+	lastTransitionTime := time.Now()
+	var reason string
+	if len(s) == 1 {
```

Comment:
To make it optional, because most of the time there is no error message when creatin a `ResourceCleanupStatus` entity and we would just pass `nil`s here.

Reply:
I think Reason should always have a value, there is always a reason why it moved from the previous state to this state even  if its just "CompletedCleanup".